lib/phraseapp_updater/phraseapp_api.rb

```diff
@@ -238,17 +238,30 @@ def store_parent_commit(commit_hash)
       phraseapp_request(Phrase::TagsApi,:tag_create, @project_id, params)
     end
 
-    def wrap_phrase_errors
-      yield
-    rescue Phrase::ApiError => e
-      if e.code == 401
-        raise BadAPIKeyError.new(e)
-      elsif e.message.match?(/not found/)
-        raise BadProjectIDError.new(e, @project_id)
-      elsif e.message.match?(/has already been taken/)
-        raise ProjectNameTakenError.new(e)
-      else
-        raise
+    def wrap_phrase_errors(retries: 10)
+      begin
+        yield
+      rescue Phrase::ApiError => e
+        if e.code == 401
+          raise BadAPIKeyError.new(e)
+        elsif e.code == 429
+          # If we bail mid-sync, it can be expensive to recover from a partial
+          # merge. Instead, aggressively try to retry.
+          if retries >= 0
+            retries -= 1
+            STDERR.puts('Rate limited, retrying in 5...')
+            sleep(5)
+            retry
+          end
```

Comment:
Technically they return a `X-Rate-Limit-Reset` which tells you when the rate limit should be okay again 🤷 It's not clear to me if 5 seconds is sufficient or not. If the reset goes back to 100% after 5 minutes though I wouldn't see much harm in just sleeping for 5 minutes.

Reply:
5 seconds may well be insufficient, but even if each failed request is itself counted towards the rate-limit, it's still slow enough that the pool will eventually drain and we'll be able to make the request. So it might be somewhat lazy behaviour, but I think it's safe.